crates/goose-server/src/state.rs

```diff
@@ -6,13 +6,17 @@ use std::path::PathBuf;
 use std::sync::atomic::AtomicUsize;
 use std::sync::Arc;
 use tokio::sync::Mutex;
+use tokio_util::sync::CancellationToken;
+
 #[derive(Clone)]
 pub struct AppState {
     pub(crate) agent_manager: Arc<AgentManager>,
     pub recipe_file_hash_map: Arc<Mutex<HashMap<String, PathBuf>>>,
     pub session_counter: Arc<AtomicUsize>,
     /// Tracks sessions that have already emitted recipe telemetry to prevent double counting.
     recipe_session_tracker: Arc<Mutex<HashSet<String>>>,
+    /// Tracks cancellation tokens for active sessions
+    pub(crate) session_cancellation_tokens: Arc<Mutex<HashMap<String, CancellationToken>>>,
```

Comment:
this seems a bit confusing; do we want to keep track of the active sessions in the app state or in the session manager?